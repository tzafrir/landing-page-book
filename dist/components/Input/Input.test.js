'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _enzyme = require('enzyme');

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

var _Input = require('./Input');

var _Input2 = _interopRequireDefault(_Input);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });

describe('Input', function () {
  var props = void 0;
  var mounted = void 0;
  var input = function input() {
    if (!mounted) {
      mounted = (0, _enzyme.shallow)(_react2.default.createElement(_Input2.default, props));
    }
    return mounted;
  };
  describe('Input', function () {
    beforeEach(function () {
      props = {};
      mounted = undefined;
    });

    it('should render', function () {
      expect((0, _enzyme.shallow)(_react2.default.createElement(_Input2.default, props))).toMatchSnapshot();
    });

    it('always renders a div', function () {
      var divs = input().find('input');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});