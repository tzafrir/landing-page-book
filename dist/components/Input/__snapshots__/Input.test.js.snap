// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className=""
    icon=""
    iconAlign="left"
    id=""
    label=""
    labelColon={false}
    name=""
    onChange={[Function]}
    placeholder=""
    size=""
    type="text"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "",
        <div
          className="Input__Container"
        >
          <input
            className="form-control  "
            id=""
            name=""
            onChange={[Function]}
            placeholder=""
            type="text"
            value=""
          />
          
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-control  "
              id=""
              name=""
              onChange={[Function]}
              placeholder=""
              type="text"
              value=""
            />,
            false,
            "",
          ],
          "className": "Input__Container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control  ",
              "id": "",
              "name": "",
              "onChange": [Function],
              "placeholder": "",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          false,
          "",
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "",
          <div
            className="Input__Container"
          >
            <input
              className="form-control  "
              id=""
              name=""
              onChange={[Function]}
              placeholder=""
              type="text"
              value=""
            />
            
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form-control  "
                id=""
                name=""
                onChange={[Function]}
                placeholder=""
                type="text"
                value=""
              />,
              false,
              "",
            ],
            "className": "Input__Container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control  ",
                "id": "",
                "name": "",
                "onChange": [Function],
                "placeholder": "",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            false,
            "",
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
